mos-platform/geos-cbm/geos_crt.c

```diff
@@ -0,0 +1,2013 @@
+  __r4 = right;
+  __r2H = bottom;
+  asm volatile("jsr __Rectangle " : : : "a","x","y","c","v");
+}
+
+void FrameRectangle(uint16_t left, uint8_t top, uint16_t right, uint8_t bottom, uint8_t pattern)
+{
+  __r3 = left;
+  __r2L = top;
+  __r4 = right;
+  __r2H = bottom;
+  asm volatile("jsr __FrameRectangle" : "=a"(pattern) : "a"(pattern) : "x","y","c","v");
+}
+
+void InvertRectangle(uint16_t left, uint8_t top, uint16_t right, uint8_t bottom)
+{
+  __r3 = left;
+  __r2L = top;
+  __r4 = right;
+  __r2H = bottom;
+  asm volatile("jsr __InvertRectangle " : : : "a","x","y","c","v");
+}
+
+void RecoverRectangle(uint16_t left, uint8_t top, uint16_t right, uint8_t bottom)
+{
+  __r3 = left;
+  __r2L = top;
+  __r4 = right;
+  __r2H = bottom;
+  asm volatile("jsr __RecoverRectangle " : : : "a","x","y","c","v");
+}
+
+void ImprintRectangle(uint16_t left, uint8_t top, uint16_t right, uint8_t bottom)
+{
+  __r3 = left;
+  __r2L = top;
+  __r4 = right;
+  __r2H = bottom;
+  asm volatile("jsr __ImprintRectangle " : : : "a","x","y","c","v");
+}
+
+void BitmapUp(const uint8_t *bitmap, uint8_t card_left, uint8_t top, uint8_t card_width, uint8_t height)
+{
+  __r0 = (uint16_t)bitmap;
+  __r1L = card_left;
+  __r1H = top;
+  __r2L = card_width;
+  __r2H = height;
+  asm volatile("jsr __BitmapUp " : : : "a","x","y","c","v");
+}
+
+void BitmapClip(const uint8_t *bitmap, uint8_t card_left, uint8_t top, uint8_t card_width, uint8_t height,
+                uint8_t left_skip, uint8_t right_skip, uint16_t top_skip)
+{
+  __r0 = (uint16_t)bitmap;
+  __r1L = card_left;
+  __r1H = top;
+  __r2L = card_width;
+  __r2H = height;
+
+  __r11L = left_skip;
+  __r11H = right_skip;
+  __r12 = top_skip;
+
+  asm volatile("jsr __BitmapClip " : : : "a","x","y","c","v");
+}
+
+void BitOtherClip(const uint8_t *bitmap, uint8_t card_left, uint8_t top, uint8_t card_width, uint8_t height,
+                  uint8_t left_skip, uint8_t right_skip, uint16_t top_skip,
+                  vector app_input, vector sync)
+{
+  __r0 = (uint16_t)bitmap;
+  __r1L = card_left;
+  __r1H = top;
+  __r2L = card_width;
+  __r2H = height;
+
+  __r11L = left_skip;
+  __r11H = right_skip;
+  __r12 = top_skip;
+  __r13 = (uint16_t)app_input;
+  __r14 = (uint16_t)sync;
+
+  asm volatile("jsr __BitOtherClip " : : : "a","x","y","c","v");
+}
+
+void GraphicsString(const uint8_t *graph_string)
+{
+  __r0 = (uint16_t)graph_string;
+  asm volatile("jsr __GraphicsString " : : : "a","x","y","c","v");
+}
+
+void GetScanLine(uint8_t y, uint16_t *first_screen_byte, uint16_t *first_backgr_byte)
+{
+  asm volatile("jsr __GetScanLine" : : "x"(y) : "a","c","v");
+  *first_screen_byte = __r5;
+  *first_backgr_byte = __r6;
+}
+
+/***********************************************************************
+ * GEOS Character routines
+ ***********************************************************************/
+
+void PutString(uint16_t x, uint8_t y, const char *str)
+{
+  __r11 = x;
+  __r1H = y;
+  __r0 = (uint16_t)str;
```

Comment:
Is it OK to overwrite `__r0`, or should it be added to the clobber list? I do not fully understand the custom register setup, so it may be perfectly fine.